### ISSUE TITLE
Question on Temperature Control

### ISSUE THREAD

**Issue by steowens** [MarlinFirmware/Marlin#4964]
Rather than having the main board handle the temperature sensing and control, what would be the feasibility of offloading sensing of individual thermistors to a sparate microcontroller using fore example the 3 dollar arduino pro mini controllers and perhapse a high fidelity 12 bit ADC controller.  Perhaps just having the marlin read the temperature from the serial input or I2C or something instead of directly managing the thermistor interface?

**Comment by Grogyan** [comment]
It is highly feasible, but not worthwhile, as an arduino mini is a high added cost, vs an off board 12bit or higher ADC.
At which point you might as well use a thermocouple, which is much more linear than thermistors.

3D printer control boards are going through a slow transition to 32bit microprocessors, which have higher resolution ADCs on board.
The hope here is that RAMPS like boards will be deprecated and not really worthwhile in the long term

**Comment by thinkyhead** [comment]
@steowens Marlin now has a nice i2c interface, which I've used in one project – to make a 2nd RAMPS do all the work of moving, homing, and leveling a Z axis with 4 individual steppers. I found i2c was fast enough, and I had a very simple protocol. I described my implementation in this thread: https://github.com/MarlinFirmware/Marlin/issues/4776#issuecomment-246262879

It should be relatively easy to utilize a second instance of Marlin to manage the temperatures, the LCD controller, or any other things that a second controller could take on, only relaying messages back to a main controller as needed. It would be most straightforward to have one controller manage the high-level stuff, user interface, GCode parser, wifi, etc., while the slave controller focused only on moving the steppers, managing the heaters, etc.
